tests/unit/util/recalls.unit.test.ts

```diff
@@ -0,0 +1,238 @@
+      expect(res).toStrictEqual(emptyResponse);
+    });
+    it('SHOULD filter to nothing when all vehicles are fixed', () => {
+      const recallResponse = {
+        vin: '1234',
+        manufacturer: 'test manufacturer',
+        recalls: [
+          {
+            manufacturerCampaignReference: '123ABC',
+            dvsaCampaignReference: '1234ABC',
+            recallCampaignStartDate: new Date((date.setDate(date.getDate() - 1))).toISOString().split('T')[0],
+            repairStatus: 'FIXED',
+          },
+          {
+            manufacturerCampaignReference: '123CBA',
+            dvsaCampaignReference: '1234CBA',
+            recallCampaignStartDate: new Date((date.setDate(date.getDate() - 1))).toISOString().split('T')[0],
+            repairStatus: 'FIXED',
+          },
+        ],
+        lastUpdatedDate: '123456',
+      };
+      const res = RecallUtils.filterMotRecalls(recallResponse as MotRecalls);
+      expect(res).toStrictEqual(emptyResponse);
+    });
+    it('SHOULD filter to nothing when all vehicles are fixed and all dates are in the future', () => {
+      const recallResponse = {
+        vin: '1234',
+        manufacturer: 'test manufacturer',
+        recalls: [
+          {
+            manufacturerCampaignReference: '123ABC',
+            dvsaCampaignReference: '1234ABC',
+            recallCampaignStartDate: new Date((date.setDate(date.getDate() + 1))).toISOString().split('T')[0],
+            repairStatus: 'FIXED',
+          },
+          {
+            manufacturerCampaignReference: '123CBA',
+            dvsaCampaignReference: '1234CBA',
+            recallCampaignStartDate: new Date((date.setDate(date.getDate() + 1))).toISOString().split('T')[0],
+            repairStatus: 'FIXED',
+          },
+        ],
+        lastUpdatedDate: '123456',
+      };
+      const res = RecallUtils.filterMotRecalls(recallResponse as MotRecalls);
+      expect(res).toStrictEqual(emptyResponse);
+    });
+  });
+  describe('getBearerToken', () => {
+    it('SHOULD return undefined and log an error message when it fails to retrieve bearer token', async () => {
+      const mockErr = 'test failed successfully';
+      global.fetch = jest.fn().mockImplementation(() => {
+        throw Error(mockErr);
+      });
+
+      const res = await RecallUtils.getBearerToken({} as MotSecret);
+      expect(logger.error).toHaveBeenCalledWith(`Failed to get bearer token: Error: ${mockErr}`);
+      expect(res).toBeUndefined();
+    });
+    it('SHOULD return the token upon successful retrieval of bearer token', async () => {
+      global.fetch = jest.fn().mockImplementation(() => ({
+        status: 200,
+        json: () => ({
+          access_token: 'test',
+        }),
+      } as unknown as Response));
+
+      const res = await RecallUtils.getBearerToken({} as MotSecret);
+      expect(res).toBe('test');
+    });
+  });
+  describe('getMotRecallsByVin', () => {
```

Comment:
Can you add testcase to test what is returned from the catch block of this method please.

Lines 76-77
```
  } catch (err) {
    logger.error(`failed calling MOT endpoint: Error: ${(err as Error).message}`);
    return undefined;
  }
```